packages/contracts-sdk/src/implementations/farcaster/FarcasterBundlerContract.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  EVMContractAddress,
+  BlockchainCommonErrors,
+  FarcasterBundlerContractError,
+} from "@snickerdoodlelabs/objects";
+import { ethers } from "ethers";
+import { injectable } from "inversify";
+import { ResultAsync } from "neverthrow";
+
+import { BaseContract } from "@contracts-sdk/implementations/BaseContract.js";
+import { IEthersContractError } from "@contracts-sdk/implementations/BlockchainErrorMapper.js";
+import {
+  ContractOverrides,
+  IFarcasterBundlerContract,
+  WrappedTransactionResponse,
+} from "@contracts-sdk/interfaces/index.js";
+import { RegistrationParams } from "@contracts-sdk/interfaces/objects/farcaster/RegistrationParams";
+import { SignerParams } from "@contracts-sdk/interfaces/objects/farcaster/SignerParams";
+import { ContractsAbis } from "@contracts-sdk/interfaces/objects/index.js";
+
+@injectable()
+export class FarcasterBundlerContract
+  extends BaseContract<FarcasterBundlerContractError>
+  implements IFarcasterBundlerContract
+{
+  constructor(
+    protected providerOrSigner: ethers.Provider | ethers.Signer,
+    protected contractAddress: EVMContractAddress,
```

Comment:
The contract addresses for the farcaster contracts are known- I think we can omit this parameter and bake it into the constructor here. Also, I think we should do a check, and make sure that providerOrSigner is for Optimism. It should throw an error if you give a providerOrSigner that is not for the correct chain.